### PR TITLE
Simplify header handling (attempt #2)

### PR DESCRIPTION
**Pull request by Acconut**
This PR superseeds https://github.com/tus/tus-v2/pull/6.

The included changes are:
- Include the Upload-Incomplete header in HEAD responses, to allow easy identification of an upload's status.
- Clarify when an upload is complete and when not.
- Drop requirement that Upload-Incomplete header is always truthy if present.
- Change response code for mismatching Upload-Offset from general 400 to specific 409.
- Add brief description what each header value represents.

### CODE REVIEW

File: draft-tus-resumable-uploads-protocol.md [hunk 1]
```diff
@@ -248,7 +248,7 @@ The request MUST use the `HEAD` method and include the `Upload-Token` header. Th
 
 The client MUST NOT perform the Offset Retrieving Procedure ({{offset-retrieving}}) while the Upload Transfer Procedures ({{upload-transfer}}) is in progress.
 
-If the server considers the upload associated with this token active, it MUST send back a `204 (No Content)` response with a header `Upload-Offset` which indicates the resumption offset for the client.
+If the server has resources allocated for this token, it MUST send back a `204 (No Content)` response. The response MUST include the `Upload-Offset` header set to the current resumption offset for the client. The response MUST include the `Upload-Incomplete` header which is set to true if and only if the upload is incomplete. An upload is considered complete if and only if the server completely and succesfully received a corresponding Upload Transfer Procedure  ({{upload-transfer}}) request with a falsy `Upload-Incomplete` header.
```
Comment (smatsson) [inline draft-tus-resumable-uploads-protocol.md]: >If the server considers the upload associated with this token active, 

Should probably still be there from #26

Comment (justinruggles) [reply draft-tus-resumable-uploads-protocol.md]: "falsy" is a bit vague

Comment (guoye-zhang) [reply draft-tus-resumable-uploads-protocol.md]: Add brief explanation that the server may deactivate the token immediately after the transfer completes, so it won't ever respond with `Upload-Incomplete: ?0`. We are not forcing servers to keep the completed upload around.

Comment (Acconut) [reply draft-tus-resumable-uploads-protocol.md]: Thank you all for the feedback!

**Pull request merged**